solidity/dashboard/src/App.js

```diff
@@ -16,19 +16,21 @@ const App = () => (
       <Web3ContextProvider>
         <ModalContextProvider>
           <ContractsDataContextProvider>
-            <SideMenuProvider>
-              <Router>
-                <main>
-                  <Header />
-                  <aside>
+            <HeaderContextProvider>
+              <SideMenuProvider>
```

Comment:
yeah 100% right. let's drop a side menu context.

I decided to remove `MenuButton` component from a header because we don't have any mobile views on Figma yet and I think it's more reasonable to deal with mobile views in a separate PR.

Regarding the contexts, its on my plate too, to cleanup this mess.